src/packet_analysis/protocol/pppoe/PPPoE.cc

```diff
@@ -12,8 +12,20 @@ bool PPPoEAnalyzer::AnalyzePacket(size_t len, const uint8_t* data, Packet* packe
         return false;
     }
 
+    size_t payload_length = (data[4] << 8u) + data[5];
     // Extract protocol identifier
```

Comment:
```suggestion
```

Reply:
:man_shrugging:  - either that, or adding a comment to the payload_length extraction, but that seems to document just what's done.